nextcord/application_command.py

```diff
@@ -111,6 +117,48 @@ def _cog_special_method(func: FuncT) -> FuncT:
     return func
 
 
+class InteractionDefault:
+    # TODO: fix docstring lol
+    """A class that makes all thing default.
+
+    To make a default from the :class:`Interaction`, you can do like this:
+    .. code-block:: python3
+        class InteractionUser(InteractionDefault):
+            def default(self, interaction: Interaction):
+                return interaction.user
+
+        @bot.slash_command()
+        async def balance(interaction, user = SlashOption(name="user", description="The user to check the balance", default=InteractionUser):
+            ...
+    """
+
+    def default(self, interaction: Interaction) -> None:
+        raise NotImplementedError
+
+
+class InteractionUser(InteractionDefault):
+    """Return the author of the interaction."""
+
+    def default(self, interaction: Interaction) -> Optional[Union[Member, User]]:
+        return interaction.user
+
+
+class InteractionGuild(InteractionDefault):
+    """Return the guild of this interaction."""
+
+    def default(self, interaction: Interaction) -> Optional[Guild]:
+        return interaction.guild
+
+
+class InteractionChannel(InteractionDefault):
+    """Return the channel that the interaction was executed in."""
+
+    def default(
+        self, interaction: Interaction
+    ) -> Optional[Union[GuildChannel, PartialMessageable, Thread]]:
```

Comment:
this does not include all possibilities - such as `DMChannel`, might be best to use `abc.Messageable`